src/openms/source/ANALYSIS/TOPDOWN/PeakGroup.cpp

```diff
@@ -313,41 +324,79 @@ namespace OpenMS
       int c_zero = 0;
       for (int z = max_sig_charge; z <= t_nmax_abs_charge; z++)
       {
-        if (per_charge_int_[z] <= 0)
+        if (per_charge_signal_pwr_[z] <= 0)
         {
           c_zero++;
         }
         else
         {
           c_zero = 0;
+          nmax_abs_charge = z;
         }
         if (c_zero > 2)
         {
           break;
         }
-        nmax_abs_charge = z;
+
       }
       c_zero = 0;
       for (int z = max_sig_charge; z >= t_nmin_abs_charge; z--)
       {
-        if (per_charge_int_[z] <= 0)
+        if (per_charge_signal_pwr_[z] <= 0)
         {
           c_zero++;
         }
         else
         {
           c_zero = 0;
+          nmin_abs_charge = z;
         }
         if (c_zero > 2)
         {
           break;
         }
-        nmin_abs_charge = z;
       }
     }
+    if(max_abs_charge_ != nmax_abs_charge || min_abs_charge_ != nmin_abs_charge)
+    {
+      std::vector<LogMzPeak> new_sig;
+      std::vector<LogMzPeak> new_noise;
+      new_sig.reserve(size());
+      new_noise.reserve(noisy_peaks_.size());
+
+      for(auto p:logMzpeaks_)
```

Comment:
```suggestion
      for (const auto& p : logMzpeaks_)
```
LogMzPeak is quite large, right?